x-pack/plugins/observability/public/hooks/use_time_range.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { parse } from 'query-string';
+import { useLocation } from 'react-router-dom';
+import { TimePickerTime } from '../components/shared/data_picker';
+import { getAbsoluteTime } from '../utils/date';
+import { UI_SETTINGS, useKibanaUISettings } from './use_kibana_ui_settings';
+import { usePluginContext } from './use_plugin_context';
+
+const getParsedParams = (search: string) => {
+  return search ? parse(search[0] === '?' ? search.slice(1) : search, { sort: false }) : {};
```

Comment:
`search` is required so I think you can drop the ternary. Also, `search` should always start with "?" so I don't see the need for that check

```suggestion
  return parse(search.slice(1), { sort: false });
```